### ISSUE TITLE
Can we reduce special logic in index.js even more?

### ISSUE THREAD

**Issue by bsudekum** [Project-OSRM/osrm-text-instructions#6]
The smaller this logic is, the more extensible this library will be. We should try to see if we can reduce this logic even further.

/cc @freenerd

**Comment by 1ec5** [comment]
#23 muddies up the format even further: now the `[optional]` syntax can be replaced outright with a `destination` "template". For localization purposes, that PR should've created a separate set of messages for each towards case instead of that "template".

**Comment by freenerd** [comment]
When I implemented #23 my natural way to go would have actually been named `[]` variables. That seems complex though and ripe for weird string manipulation cases (e.g. setting and retaining spaces correctly). Instead, we could also remove the `[]` variables completely, by splitting every `turn/modifier` combination into a new object with several cases.

Now:

```
        "off ramp": {
            "defaultInstruction": "Take the ramp[ onto {way_name}]",
         }
```

New:

```
        "off ramp": {
            "defaultInstruction":  {
                 "default": "Take the ramp",
                 "name": "Take the ramp onto {way_name}",
                 "destination": "Take the ramp towards {destination_name}",
            }
         }
```
